### PR TITLE
A* Implementation for Pathfinding

### PR DESCRIPTION
**Pull request by Nick-Trot**
pepeDance

### CODE REVIEW

File: src/Testing/main.cpp [hunk 1]
```diff
@@ -1,18 +1,42 @@
...
+	PathfindingMap exampleMap{ {
+		{0, {1}},
+		{1, {0,2,4}},
+		{2, {1,3}},
+		{3, {2} },
+		{4, {1,5,8} },
+		{5, {4,6} },
+		{6, {5,7} },
+		{7, {6,8} },
+		{8, {4,7} },
```
Comment (mattskrobola) [inline src/Testing/main.cpp:31]: nit: can be in the format of ` {a.id, {b.id}}, {b.id, {a.id,c.id,e.id}},` ect

File: src/Engine/GameObject/Pathfinding.cpp [hunk 2]
```diff
@@ -1,26 +1,96 @@
...
+std::vector<UID_t> PathfindingMap::Neighbors(UID_t id) {
+	return edges[id];
+}
 
+std::unordered_map<UID_t, UID_t> pathSearch(PathfindingMap map, UID_t start, UID_t goal) {
+	PriorityQueue<UID_t, priority_t> frontier;
+	frontier.put(start, 0);
+	std::unordered_map<UID_t, UID_t> cameFrom;
+	cameFrom[start] = start;
+	std::unordered_map<UID_t, double> costSoFar;
```
Comment (mattskrobola) [inline src/Engine/GameObject/Pathfinding.cpp:44]: nit: could we name this curLowestCost?

File: src/Engine/GameObject/Pathfinding.cpp [hunk 3]
```diff
@@ -1,26 +1,96 @@
...
+
+std::vector<UID_t> PathfindingMap::Neighbors(UID_t id) {
+	return edges[id];
+}
 
+std::unordered_map<UID_t, UID_t> pathSearch(PathfindingMap map, UID_t start, UID_t goal) {
+	PriorityQueue<UID_t, priority_t> frontier;
+	frontier.put(start, 0);
+	std::unordered_map<UID_t, UID_t> cameFrom;
+	cameFrom[start] = start;
```
Comment (mattskrobola) [inline src/Engine/GameObject/Pathfinding.cpp:43]: nit: could we name this prev / prevNode?

File: src/Engine/GameObject/Pathfinding.cpp [hunk 4]
```diff
@@ -1,26 +1,96 @@
 #include "Pathfinding.hpp"
+#include "Scene\Scene.hpp"
 
-std::vector<char> PathfindingMap::Neighbors(char id) {
-	return edges[id];
+#define MAXPATHNODE 1024
```
Comment (mattskrobola) [inline src/Engine/GameObject/Pathfinding.cpp:4]: Will we ever have more than 1024 Nodes in a path?

**Review by mattskrobola — COMMENTED** [review]
Great work 💯! From a high level LGTM just added some nits + questions but nothing blocking! Will let @bitGraf give the final signoff as I don't have much context on this project! 😄

File: src/Engine/GameObject/Pathfinding.cpp (hunk 4 above)
Comment (Nick-Trot) [reply src/Engine/GameObject/Pathfinding.cpp:4]: No

File: src/Engine/GameObject/Pathfinding.cpp (hunk 3 above)
Comment (Nick-Trot) [reply src/Engine/GameObject/Pathfinding.cpp:43]: No

File: src/Engine/GameObject/Pathfinding.cpp (hunk 2 above)
Comment (Nick-Trot) [reply src/Engine/GameObject/Pathfinding.cpp:44]: Yes

**Comment by Nick-Trot** [comment]
What's done is done..

**Pull request merged**